File: C++/2.FiboEvenSumOneFunc.cpp
#include <iostream>
#include <vector>

// Function to find the sum of even Fibonacci numbers under a given limit
int fibonacciSumEven(int limit) {
    std::vector<int> fibonacciSequence = {0, 1};
    int evenSum = 0;

    // Main loop to generate Fibonacci numbers
    while (true) {
        // Calculate the next Fibonacci number
        int nextNumber = fibonacciSequence.back() + fibonacciSequence[fibonacciSequence.size() - 2];

        // Check if the next Fibonacci number exceeds the limit
        if (nextNumber >= limit) {
            break;
        }

        // Check if the next Fibonacci number is even
        if (nextNumber % 2 == 0) {
            evenSum += nextNumber;  // Add the even Fibonacci number to the sum
        }

        // Append the next Fibonacci number to the sequence
        fibonacciSequence.push_back(nextNumber);
    }

    return evenSum;  // Return the sum of even Fibonacci numbers
}

int main() {
    // Example: Find the sum of even Fibonacci numbers under 4 million
    int result = fibonacciSumEven(4000000);
    std::cout << result << std::endl;

    return 0;
}

File: Pyhton/2.FiboEvenSumOneFunc.py
# Function to find the sum of even Fibonacci numbers under a given limit
def fibonacci_sum_even(limit):
    # Initialize variables
    fibonacci_sequence = [0, 1]
    even_sum = 0

    # Main loop to generate Fibonacci numbers
    while True:
        # Calculate the next Fibonacci number
        next_number = fibonacci_sequence[-1] + fibonacci_sequence[-2]

        # Check if the next Fibonacci number exceeds the limit
        if next_number >= limit:
            break

        # Check if the next Fibonacci number is even
        if next_number % 2 == 0:
            even_sum += next_number  # Add the even Fibonacci number to the sum

        # Append the next Fibonacci number to the sequence
        fibonacci_sequence.append(next_number)

    return even_sum  # Return the sum of even Fibonacci numbers

# Example: Find the sum of even Fibonacci numbers under 4 million
result = fibonacci_sum_even(4000000)
print(result)

